sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileSourceStrategy.scala

```diff
@@ -97,7 +98,16 @@ object FileSourceStrategy extends Strategy with Logging {
         dataColumns
           .filter(requiredAttributes.contains)
           .filterNot(partitionColumns.contains)
-      val outputSchema = readDataColumns.toStructType
+      val outputSchema = if (fsRelation.sqlContext.conf.isParquetNestColumnPruning) {
```

Comment:
It's my mistake here, I can only make sure this patch work for parquet,so I should check the fileFormat here, also like the config namespace(`spark.sql.parquet.nestColumnPruning`), it can only work for parquet. I add a patch to fix this.